service/iot-service-samples/pnp-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/Thermostat.java

```diff
@@ -0,0 +1,60 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.service.devicetwin.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class Thermostat {
+    private static final String propertyName = "targetTemperature";
+    private static final double propertyValue = 75.5;
+    private static final String methodName = "reboot";
+    private static final Long responseTimeout = TimeUnit.SECONDS.toSeconds(200);
+    private static final Long connectTimeout = TimeUnit.SECONDS.toSeconds(5);
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("Starting sample...");
+
+        // get user inputs from environment variables
+        String iotHubConnectionString = System.getenv("IOTHUB_CONNECTION_STRING");
+        String deviceId = System.getenv("DEVICE_ID");
+
+        // connect to the IotHub
+        DeviceTwin twinClient = DeviceTwin.createFromConnectionString(iotHubConnectionString);
+        DeviceTwinDevice device = new DeviceTwinDevice(deviceId);
+
+        // get the device twin
+        System.out.println("Getting the Device twin");
+        twinClient.getTwin(device);
+        // display the pnp model id property of the device
+        System.out.println("The pnp model id is: " + device.getModelId());
+
+        // update the device twin properties
+        Set<Pair> desiredProperties = new HashSet<Pair>();
+        desiredProperties.add(new Pair(propertyName, propertyValue));
+        device.setDesiredProperties(desiredProperties);
+        System.out.println("Updating Device twin (targetTemperature, 75)");
+        twinClient.updateTwin(device);
+        System.out.println("The targetTemperature is set to: " + device.getDesiredProperties().iterator().next().getValue());
+
+        // invoke command on the device
+        System.out.println("Invoking Reboot command");
+        DeviceMethod methodClient = DeviceMethod.createFromConnectionString(iotHubConnectionString);
+        Map<String, Object> payload = new HashMap<String, Object>()
+        {
+            {
+                put("delay", 1);
```

Comment:
It would be good to add way to create payload based on user input as you have for methodname

Reply:
I will do this in next PR when we change the payload to match the new one.